src/pages/ShoppingList.js

```diff
@@ -57,6 +43,23 @@ const ShoppingList = ({ token }) => {
         );
     };
 
+    const filterShoppingListByTimeframe = (shoppingListArray) => {
+        const alphabeticalSort = (a,b) => {
+            const aName = normalizeString(a.itemName);
+            const bName = normalizeString(b.itemName);
+            if (aName < bName) {return -1;}
+            if (aName > bName) {return 1;}
+            return 0;
+        }
+        const seven = shoppingListArray.filter(item => item.timeFrame == 7).sort(alphabeticalSort);
+        const fourteen = shoppingListArray.filter(item => item.timeFrame == 14).sort(alphabeticalSort);
+        const thirty = shoppingListArray.filter(item => item.timeFrame == 30).sort(alphabeticalSort);
+        
+        //does not consider a case where timeFrame is none of these
+
+        return seven.concat(fourteen).concat(thirty);
+    };
```

Comment:
V. cool!!!!  I think you could also do `return [...seven, ...fourteen, ...thirty]`. Just an alternative, though.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax